_posts/2019-06-19-upgrade-rails-101-the-roadmap-to-smooth-upgrades.markdown

```diff
@@ -0,0 +1,59 @@
+---
+layout: post
+title: "Upgrade Rails 101: The Roadmap to Smooth Upgrades"
+date: 2019-06-19 10:00:00
+categories: ["RailsConf"]
+authors: ["etagwerker"]
+---
+
+This year's [RailsConf](https://railsconf.com) was a special conference for me.
+It was my third time attending and my first time speaking at the conference. I
+conducted a 2-hour workshop for anyone interested in upgrading their Rails
+application: [Upgrade Rails 101: The Roadmap to Smooth Upgrades](https://railsconf.com/program/workshops#session-776)
+
+Here are a few lessons learned from running such an ambitious workshop.
+
+<!--more-->
+
+## Scope
+
+I believe the scope was way too ambitious. I pitched the workshop as an opportunity
+to bring *any* Rails application and leave with a [roadmap to upgrade](https://fastruby.io/roadmap)
+to the next version of Rails. As if that had not been enough, I also told attendees
+that they could use a sample, open source application to practice our steps to
+upgrade.
+
+I should have stuck to the sample application. As the workshop went on, I found
+myself addressing issues regarding random applications **and** the sample
+application.
+
+## All Environments Are Different
+
+When starting the set of exercises, there was a slowdown addressing environment
+issues. There is always something slightly off about someone's environment that
+throws a wrench in the works.
+
+This could have been much simpler providing a `Dockerfile` and requiring attendees
+to have [`Docker`](https://www.docker.com) installed. That way, the setup would
+have been something like "just run `docker-compose up`" and done.
+
+## More Details
+
+When describing the steps to create the Rails upgrade roadmap, I should have
+provided even more detail for every step. For people that were stuck in a step,
+I could have had branches on my sample Git repository. For each step, I could
+have had a checklist to make sure that the attendees got to a stage that would
+not block them moving forward.
+
+## Reference
```

Comment:
@etagwerker Maybe before this you can add a paragraph to turn the lessons learned into a positive thing, or maybe mention some of the things that worked well. Just to make the reader want to go to the reference links that you mention below. Does that make sense?

Reply:
@lubc sure, done!